### PR TITLE
[Feature][Connector-v2] Neo4j sink connector

### PR DESCRIPTION
**Pull request by getChan**
## Purpose of this pull request

To add Neo4j Sink Connector

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
hi, could you add the license header?

**Comment by getChan** [comment]
> hi, could you add the license header?

oh. done.

**Review by EricJoy2048 — COMMENTED** [review]
LGTM
Thank you for your contribute. I think Neo4j Sink can support 2PC submit, If you are interested, you can implement it in the future.

File: seatunnel-connectors-v2/connector-neo4j/pom.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.neo4j.driver</groupId>
+            <artifactId>neo4j-java-driver</artifactId>
+            <version>4.4.9</version>
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-neo4j/pom.xml]: We manage versions uniformly in the root pom

File: seatunnel-connectors-v2/connector-neo4j/src/main/java/org/apache/seatunnel/connectors/seatunnel/neo4j/config/Neo4jConfig.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.Map;
+
+@Getter
+@Setter
+public class Neo4jConfig implements Serializable {
+
+    public static final String KEY_SINK_PLUGIN_NAME = "Neo4jSink";
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-neo4j/src/main/java/org/apache/seatunnel/connectors/seatunnel/neo4j/config/Neo4jConfig.java]: It's better used Neo4j

Comment (getChan) [reply seatunnel-connectors-v2/connector-neo4j/src/main/java/org/apache/seatunnel/connectors/seatunnel/neo4j/config/Neo4jConfig.java:30]: I think so.
but there's already exist plugin name `Neo4j` which is V1 neo4j Connector.

File: seatunnel-connectors-v2/connector-neo4j/pom.xml (hunk 1 above)
Comment (getChan) [reply seatunnel-connectors-v2/connector-neo4j/pom.xml]: I fixed it

File: seatunnel-connectors-v2/connector-neo4j/src/main/java/org/apache/seatunnel/connectors/seatunnel/neo4j/config/Neo4jConfig.java (hunk 2 above)
Comment (getChan) [reply seatunnel-connectors-v2/connector-neo4j/src/main/java/org/apache/seatunnel/connectors/seatunnel/neo4j/config/Neo4jConfig.java]: I fixed it

**Review by CalvinKirs — APPROVED** [review]

**Comment by CalvinKirs** [comment]
Thanks for your contribution. looking forward you next PR:)

**Pull request merged**